### ISSUE TITLE
Intellisense still not fully working with Enum classes

### ISSUE THREAD

**Issue by tomhsel** [microsoft/pylance-release#4928]
I believe this may be related to #1905 and #4734.

## Environment data

-   Language Server version: 2023.9.30
-   OS and version: Windows 11 build 22621.2283 and Ubuntu 22.04.3 (WSL)
-   Python version (& distribution if applicable, e.g. Anaconda):
    - 3.8.10, 3.9.13, 3.10.11, 3.11.6 (python.org Windows installers)
    - 3.10.12 (through apt on Ubuntu)

## Code Snippet

```python
from enum import Enum

class Test(Enum):
    X = "x"

    @property
    def is_x(self) -> bool:
        return True
```

## Repro Steps

1. Type `Test.X.`

## Expected behavior

The property `is_x` should be displayed in the suggestions for `Test.X.`

## Actual behavior

Only attributes that are universally applicable to Enum members are displayed (`name`, `value`, and the standard sunder/dunder attributes)

Even worse, if the Enum class doesn't inherit directly from `enum.Enum`, the suggestions popup is empty (or displays `No suggestions.` if you try to force the matter with `Ctrl+Space`). This includes Enums that make use of `enum.IntEnum` or Python 3.11's `enum.ReprEnum` and `enum.StrEnum`.

The example from https://docs.python.org/3/howto/enum.html#planet also fails to produce the attributes created in `__init__`.

It should also be noted that all of these attributes added in each of these various ways are correctly understood by the typing system when used, they are simply missing from the provided suggestions.

**Comment by ion-elgreco** [comment]
I also noticed this; I am adding all kind of methods to our enums but they don't show up in autocomplete 0_o

**Comment by debonte** [comment]
This bug is a side-effect of changes made for https://github.com/microsoft/pylance-release/issues/1905 and https://github.com/microsoft/pylance-release/issues/4734. We're still filtering out too many members on enum instances.

**Comment by fordneild** [comment]
Hi @heejaechang any updates on this? I think giving enums custom methods is a pretty great pattern that I encourage folks to do, but the lack of intellisense makes this tough. many thanks :)

**Comment by heejaechang** [comment]
I believe it is already fixed. I think we forgot to mark this as fixed.

![image](https://github.com/microsoft/pylance-release/assets/1333179/fd68af7a-6173-4e48-b0ab-810bb8948b6e)

**Comment by Denis-Alexeev** [comment]
@heejaechang But it is not working. The problem, that the instnce of enum must also have this hint
![telegram-cloud-photo-size-2-5465241967882524802-x](https://github.com/microsoft/pylance-release/assets/52697534/d2038ae0-a3a3-48fe-be1f-bda3f9c9b4ae)

**Comment by heejaechang** [comment]
This issue has been fixed in prerelease version 2024.7.100, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#20247100-15-July-2024-PreRelease)
